llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -554,13 +560,12 @@ def forward(
         use_cache = (use_cache
                      if use_cache is not None else self.config.use_cache)
 
-        # if input_embeds is not none, raise a not implemented error
-        if inputs_embeds is not None:
-            raise NotImplementedError(
-                'inputs_embeds has to be None (for hf/peft support).')
-        # decoder outputs consists of (dec_features, layer_state, dec_hidden, dec_attn)
+        if input_ids is not None and inputs_embeds is not None:
+            raise ValueError(
+                'You cannot specify both input_ids and inputs_embeds.')
```

Comment:
Just realized this might be wrong. I need to look at how HF does this, but I think that during autoregressive generation we might get both of these and that's ok..?